### ISSUE TITLE
Randomness in Results

### ISSUE THREAD

**Issue by Nik-V9** [dcharatan/flowmap#42]
Hey @dcharatan, Thanks for the nice code!

I've been running the code through the overfit.py file on my own data. Even though I seed everything, the code produces different results each time (sometimes the performance delta is large).

I was wondering if there are any elements of the code which have randomness that can't be seeded? What is the best way to seed the code for benchmarking? For the paper evaluations, how do you go about this issue?

I tried setting the PyTorch Lightning Trainer to determinisitic but there were a few operations like 2D grid sampling which didn't have a determinisitc implementation.

I'm attaching snippets of my evaluation here across 6 back-to-back runs with the same command and seed:
<img width="235" alt="image" src="https://github.com/dcharatan/flowmap/assets/56788999/a1b289b3-e961-46e7-9401-378e7e90de96">
<img width="223" alt="image" src="https://github.com/dcharatan/flowmap/assets/56788999/e3e5b65f-92c5-42fe-8200-430c53099a6e">
<img width="246" alt="image" src="https://github.com/dcharatan/flowmap/assets/56788999/3228ba67-b15b-43af-acf6-46d006dfc1f7">
<img width="225" alt="image" src="https://github.com/dcharatan/flowmap/assets/56788999/959e3910-d4a0-4ea7-b2b2-6e11908520fa">
<img width="229" alt="image" src="https://github.com/dcharatan/flowmap/assets/56788999/535050cf-4cd7-4908-891f-e1b5cddf53b4">
<img width="236" alt="image" src="https://github.com/dcharatan/flowmap/assets/56788999/7a5e813d-074b-4e85-a763-16298b3cad33">

**Comment by dcharatan** [comment]
While working on the code, we weren't careful to ensure that all randomness comes from the initial seed. You could go through the code to make sure all randomness is seeded properly, but it's probably best to just run the code several times and report a mean and standard deviation in the results for benchmarking purposes.
